src/main/java/org/terasology/Lakes/LakeTreeProvider.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.Lakes;
+
+import org.terasology.core.world.CoreBiome;
+import org.terasology.core.world.generator.facets.BiomeFacet;
+import org.terasology.core.world.generator.facets.TreeFacet;
+import org.terasology.core.world.generator.trees.TreeGenerator;
+import org.terasology.core.world.generator.trees.Trees;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.utilities.procedural.Noise;
+import org.terasology.utilities.procedural.WhiteNoise;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.Facet;
+import org.terasology.world.generation.FacetBorder;
+import org.terasology.world.generation.FacetProviderPlugin;
+import org.terasology.world.generation.GeneratingRegion;
+import org.terasology.world.generation.Produces;
+import org.terasology.world.generation.Requires;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+@RegisterPlugin
+@Requires({
+        @Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = Trees.MAXRADIUS)),
+        @Facet(value = LakeFacet.class, border = @FacetBorder(sides = Trees.MAXRADIUS)),
+        @Facet(value = BiomeFacet.class, border = @FacetBorder(sides = Trees.MAXRADIUS))
+})
+@Produces(TreeFacet.class)
```

Comment:
I think instead of creating a new facet this should rather update the `TreeFacet` to remove any trees from the area, or place additional trees around the lakes in the desert, a.k.a., oasis.

Reply:
yeah, fixed this